frontend/hub/execution-environments/ExecutionEnvironmentForm.tsx

```diff
@@ -0,0 +1,333 @@
+            validate={(name: string) => validateName(name, t)}
+          />
+
+          {!isRemote && (
+            <PageFormTextInput<ExecutionEnvironmentFormProps>
+              name="namespace.name"
+              label={t('Namespace')}
+              placeholder={t('Enter a namespace name')}
+              isDisabled
+            />
+          )}
+
+          {isRemote && (
+            <>
+              <PageFormTextInput<ExecutionEnvironmentFormProps>
+                name="upstream_name"
+                label={t('Upstream name')}
+                placeholder={t('Enter a upstream name')}
+                isRequired
+              />
+
+              <PageFormAsyncSelect<ExecutionEnvironmentFormProps>
+                name="registry"
+                label={t('Registry')}
+                placeholder={t('Select registry')}
+                query={query}
+                loadingPlaceholder={t('Loading registry...')}
+                loadingErrorText={t('Error loading registry')}
+                limit={page_size}
+                valueToString={(value: ExecutionEnvironment) => value.name}
+                openSelectDialog={registrySelector}
+                isRequired
+              />
+
+              <TagsSelector tags={tagsToInclude} setTags={setTagsToInclude} mode={'include'} />
+              <TagsSelector tags={tagsToExclude} setTags={setTagsToExclude} mode={'exclude'} />
+            </>
+          )}
+
+          <PageFormTextArea<ExecutionEnvironmentFormProps>
+            name="description"
+            label={t('Description')}
+            placeholder={t('Enter a description')}
+            isDisabled={mode == 'add'}
+          />
+        </HubPageForm>
+      )}
+    </PageLayout>
+  );
+}
+
+function validateName(name: string, t: TFunction<'translation', undefined>) {
+  const regex = /^([0-9A-Za-z._-]+\/)?[0-9A-Za-z._-]+$/;
+  if (regex.test(name)) {
+    return undefined;
+  } else {
+    return t(
+      `Container names can only contain alphanumeric characters, ".", "_", "-" and a up to one "/".`
+    );
+  }
+}
+
+type ExecutionEnvironmentFormProps = {
+  name: string;
+  upstream_name: string;
+  description?: string;
+  registry: Registry;
+  namespace?: { name: string };
+};
+
+type PayloadDataType = {
+  include_tags?: string[];
+  exclude_tags?: string[];
+  name: string;
+  upstream_name: string;
+  registry: string;
+};
+
+type Registry = {
+  id: string;
+  name: string;
+};
+
+function TagsSelector(props: {
+  tags: string[];
+  setTags: (tags: string[]) => void;
+  mode: 'exclude' | 'include';
+}) {
+  const [tagsText, setTagsText] = useState<string>('');
+  const { tags, setTags, mode } = props;
+  const { t } = useTranslation();
+
+  const label = mode == 'exclude' ? t('Add tag(s) to exclude') : t('Add tag(s) to include');
+  const label2 = mode == 'exclude' ? t('Currently excluded tags') : t('Currently included tags');
+
+  const chipGroupProps = () => {
+    const count = '${remaining}';
```

Comment:
maybe keep the comment please? This looks like a bug without some kind of comment that it isn't

Reply:
(also, `chilGroupProps` should probably be a shared helper, since it's needed for any ChipGroup component to be localized)

Reply:
Explain please more in depth what should this comment says, me and Alex dont know what this is. I will add extended comment into the PR.

Reply:
Ideally we should add a chipgroup wrapper around PF providing the selected strings translated. That one was done for AWX, to avoid to have provide the translations over and over. IMO this could a future enhancement for the framework itself. Something like below.

```jsx
import React from 'react';

import { t } from '@lingui/macro';
import { ChipGroup as PFChipGroup } from '@patternfly/react-core';
import { number } from 'prop-types';

function ChipGroup({ numChips, totalChips, ...props }) {
  return (
    <PFChipGroup
      {...props}
      numChips={numChips}
      expandedText={t`Show less`}
      collapsedText={t`${totalChips - numChips} more`}
    />
  );
}

ChipGroup.propTypes = {
  numChips: number.isRequired,
  totalChips: number.isRequired,
};

export default ChipGroup;```

Reply:
Can we do this in separate PR? I would like this to be merged and to separate framework changes.